### ISSUE TITLE
json alt codes

### ISSUE THREAD

**Issue by vasilenkoalexey** [ndmitchell/hlint#293]
Looks like special symbols with --json flag converted incorrect. For example © conveted to 
`"hint":"Parse error: \169"`
But according to string description at json.org perhaps should be 
`"hint":"Parse error: \\169"`

At least,
`JSON.parse('[{"module":"","decl":"","severity":"Error","hint":"Parse error: \169"...')`
generate exception "SyntaxError: Unexpected number in JSON at position 65"

**Comment by ndmitchell** [comment]
I believe JSON requires only one slash there.

Assuming the JSON.parse fragment is in Javascript, then `\169` in a Javascript string is converted by Javascript into the copyright symbol, before it gets passed on to JSON.parse. Does that make sense? If not, can you provide a complete JSON.parse fragment I can paste into my browser and test.

**Comment by vasilenkoalexey** [comment]
Full Hlint result
`[{"module":"","decl":"","severity":"Error","hint":"Parse error: \169","file":"1.hs","startLine":970,"startColumn":1,"endLine":970,"endColumn":1,"from":"  Contact GitHub API Training Shop Blog About\n> \169 2016 GitHub, Inc. Terms Privacy Security Status Help\n","to":null,"note":[],"refactorings":"[]"}]`
I am just try parse as is, without any modifications. I got the same exception using online json parser, for example json.parser.online.fr

**Comment by ndmitchell** [comment]
First thing to note, be careful with json.parser.online.fr - it's a bit broken and doesn't follow the JSON spec. I found the http://json.parser.online.fr/beta/ works better. In particular the former doesn't allow newlines in JSON strings.

Reading the JSON spec, you are indeed right - `\169` is not a valid escape code. It should be either the copyright symbol in UTF8, or alternatively `\u00a9`. I've changed HLint to generate the latter.
